File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        String result = "";
        for (int i = 0; i < myString.length(); i++) {
            if (myString.charAt(i) == 'A') {
                result += "B";
            } else {
                result += "A";
            }
        }
        
        if (result.contains(pat)) {
            return 1;
        } else {
            return 0;
        }
    }
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

int solution(const char* myString, const char* pat) {
    char *s = malloc(sizeof(char)*strlen(myString)); 
    for (int i = 0; i < strlen(myString); i++) {
        if (myString[i] == 'A') {
            s[i] = 'B';
        } else {
            s[i] = 'A';
        }
    }
    if (strstr(s, pat)) {
        return 1; 
    } else {
        return 0; 
    }
}